sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusScope.cs

```diff
@@ -0,0 +1,627 @@
+            }
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to remove an ephemeral Service Bus namespace used as a container for queue and topic instances
+        ///   for a specific test run.
+        /// </summary>
+        ///
+        /// <param name="namespaceName">The name of the namespace to delete.</param>
+        ///
+        public static async Task DeleteNamespaceAsync(string namespaceName)
+        {
+            var subscription = TestEnvironment.ServiceBusSubscription;
+            var resourceGroup = TestEnvironment.ServiceBusResourceGroup;
+            var token = await AquireManagementTokenAsync().ConfigureAwait(false);
+
+            using (var client = new ServiceBusManagementClient(new TokenCredentials(token)) { SubscriptionId = subscription })
+            {
+                await CreateRetryPolicy().ExecuteAsync(() => client.Namespaces.DeleteAsync(resourceGroup, namespaceName)).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        ///   Creates a Service Bus scope associated with a queue instance, intended to be used in the context
+        ///   of a single test and disposed when the test has completed.
+        /// </summary>
+        ///
+        /// <param name="enablePartitioning">When <c>true</c>, partitioning will be enabled on the queue that is created.</param>
+        /// <param name="enableSession">When <c>true</c>, a session will be enabled on the queue that is created.</param>
+        /// <param name="forceQueueCreation">When <c>true</c>, forces creation of a new queue even if an environmental override was specified to use an existing one.</param>
+        /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
+        ///
+        /// <returns>The requested Service Bus <see cref="QueueScope" />.</returns>
+        ///
+        /// <remarks>
+        ///   If an environmental override was set to use an existing Service Bus queue resource and the <paramref name="forceQueueCreation" /> flag
+        ///   was not set, the existing queue will be assumed with no validation.  In this case the <paramref name="enablePartitioning" /> and
+        ///   <paramref name="enableSession" /> parameters are also ignored.
+        /// </remarks>
+        ///
+        public static async Task<QueueScope> CreateWithQueue(bool enablePartitioning,
+                                                             bool enableSession,
+                                                             bool forceQueueCreation = false,
+                                                             [CallerMemberName] string caller = "")
+        {
+            // If there was an override and the force flag is not set for creation, then build a scope
+            // for the specified queue.
+
+            if (!string.IsNullOrEmpty(TestEnvironment.OverrideQueueName) && (!forceQueueCreation))
```

Comment:
nit: extra parens around !forceQueueCreation

Reply:
Conversely, there was a missing set around the first clause, which I added.  I prefer to group explicitly, especially where negating is concerned.